web/src/scheduled_messages_feed_ui.ts

```diff
@@ -1,12 +1,14 @@
 import $ from "jquery";
+import assert from "minimalistic-assert";
 
 import render_scheduled_messages_indicator from "../templates/scheduled_messages_indicator.hbs";
 
 import * as narrow_state from "./narrow_state";
 import * as scheduled_messages from "./scheduled_messages";
+import type {ScheduledMessage} from "./scheduled_messages";
 import * as util from "./util";
 
-function get_scheduled_messages_matching_narrow() {
+function get_scheduled_messages_matching_narrow(): ScheduledMessage[] | false {
```

Comment:
So this `false` type is clearly the result of the typo/bug; it should just return an empty array in the `!is_conversation_view` case, and then you don't need to change the caller at all. So fix the bug, maybe in a prep commit, and then the TS conversion gets to be clean with no refactoring.